mage/query-modules/cpp/meta.md

```diff
@@ -0,0 +1,199 @@
+import TabItem from '@theme/TabItem';
+import RunOnSubgraph from '../../templates/_run_on_subgraph.mdx';
+
+export const Highlight = ({children, color}) => (
+<span
+style={{
+  backgroundColor: color,
+  borderRadius: '2px',
+  color: '#fff',
+  padding: '0.2rem',
+}}>
+{children}
+</span>
+);
+
+
+The **meta** module provides a set of procedures for generating metadata about the database.
+
+[![docs-source](https://img.shields.io/badge/source-util_module-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/tree/main/cpp/meta_module)
+
+| Trait               | Value                                                 |
+| ------------------- | ----------------------------------------------------- |
+| **Module type**     | <Highlight color="#FB6E00">**algorithm**</Highlight>  |
+| **Implementation**  | <Highlight color="#FB6E00">**C++**</Highlight>        |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight> |
+
+### Procedures
+
+## stats
+
+The stats procedure returns the following metadata about the graph:
+- `labelCount` ➡ number of unique labels in nodes
+- `relationshipTypeCount` ➡ number of unique relationship types (labels)
+- `nodeCount` ➡ number of nodes in the graph
+- `relationshipCount` ➡ number of relationships in the graph
+- `labels` ➡ map with the following (key, value) pairs:
+  - `label` : number_of_occurrences
+- `relationshipTypes` ➡ map with the following (key, value) pairs:
+  - `(:label)-[:relationship_type]->()` : number_of_occurrences
+  - `()-[:relationship_type]->(:label)` : number_of_occurrences
+  - `()-[:relationship_type]->()` : number_of_occurrences
+- `relationshipTypesCount` ➡ map with the following (key, value) pairs:
+  - `relationship_type` : number_of_occurrences
+- `stats` ➡ map which contains all of the above
+
+It is split into two version which return the same metadata:
+- stats_online - works in **O(1)** and requires setting up a trigger
+- stats_offline - traverses the whole graph
+
+### `stats_online(update_stats)`
+
+Retrieves the graph metadata in **O(1)** complexity. Requires setting up the following trigger:
+
+ ```cypher
+ CREATE TRIGGER meta_trigger BEFORE COMMIT EXECUTE CALL meta.update(createdObjects, deletedObjects, removedVertexProperties, removedEdgeProperties, setVertexLabels, removedVertexLabels);
+ ```
+This procedure tracks the data created/deleted/modified after the trigger was added. If you want to return the metadata about the whole graph you need to run the *stats_online* procedure with the *update_stats* flag set to true **once**. That flag will cause the procedure to traverse the whole graph to update the metadata. After that you can always run with the *update_stats* flag set to false and the procedure will return the metadata in **O(1)** complexity.
+
+
+#### Input:
+
+- `update_stats: bool (default=false)` ➡ if true traverses the whole graph to update the metadata otherwise returns the stored metadata
```

Comment:
```suggestion
- `update_stats: bool (default=false)` ➡ if true traverses the whole graph to update the metadata otherwise returns the stored metadata.
```